test/e2e/fixtures.go

```diff
@@ -165,8 +164,14 @@ func skipIfFeatureDisabled(tb testing.TB, feature featuregate.Feature, checkAgen
 	}
 }
 
-func skipIfProxyDisabled(t *testing.T) {
-	skipIfFeatureDisabled(t, features.AntreaProxy, true /* checkAgent */, false /* checkController */)
+func skipIfProxyDisabled(t *testing.T, data *TestData) {
+	agentConf, err := data.GetAntreaAgentConf()
+	if err != nil {
+		t.Fatalf("Error getting Antrea Agent config: %v", err)
+	}
+	if !*agentConf.AntreaProxy.Enable {
```

Comment:
can this be `nil`? should we also check the Feature gate in case it was explicitly disabled?

Reply:
It could be `nil`. I have added the code to check feature gate and if this pointer is `nil`.